### DISCUSSION TITLE
Scope variable $using:xxx with new value assignment

### DISCUSSION THREAD

**Discussion by timttran** [Badgerati/Pode#1019] [General]
Is there a way to assign a new value to a scope variable $using:xxxx when it's being used inside a route scriptblock?
If not, please advise if there's a way to overcome this problem?

I have a route that needs to retain a value and being recalled on subsequent calls.

**Comment by Badgerati** [comment]
Hi @timttran,

It's not possible to set a value back to a `$using:` variable unfortunately. They are parsed in scriptblock when the server starts, and the values are cached.

For what you're trying to achieve, using Pode's [shared state](https://badgerati.github.io/Pode/Tutorials/SharedState/) would be a better fit.

Hope that helps!

**Reply by timttran** [reply]
Thank you for your guidance.  I got it working using $state:xxxx

Also, after I've updated Pode to the latest version 2.7.1, I'm running into auto server restart error once I've modified and saved a source file.  Please take a look at the attached image.  The funny thing is that it runs again fine if I issue "pode start" command right after the error

![Capture](https://user-images.githubusercontent.com/25288383/196245068-78bbe744-6fb5-420c-b54e-ee7674ac2a68.PNG)

**Reply by Badgerati** [reply]
Hi @timttran,

This issue is known about (#1001), and will be fixed soon in v2.7.2 😃
